src/wp-admin/includes/ajax-actions.php

```diff
@@ -5571,7 +5571,12 @@ function wp_ajax_send_password_reset() {
 			/* translators: %s: User's display name. */
 			sprintf( __( 'A password reset link was emailed to %s.' ), $user->display_name )
 		);
+	} elseif ( is_wp_error( $results ) ) {
+		wp_send_json_success(
+			/* translators: %s: User's display name. */
+			sprintf( __( 'A password reset link was emailed to %s.' ), $user->display_name )
+		);
 	} else {
-		wp_send_json_error( $results->get_error_message() );
+		wp_send_json_error( 'Cannot retrieve user\'s password', 500);
```

Comment:
With the previous code here:
- If `retrieve_password()` returns `true`, a success is sent.
- If `retrieve_password()` returns something else, it's treated as the other documented return type: `WP_Error`, and an error is sent.
- In addition, if `retrieve_password()` is ever bugged and returns a non-`WP_Error`, the code will error out on `$results->get_error_message()`, highlighting the issue immediately. No "success" message will be sent _and_ the error will immediately alert the developer so they can rectify it. This way, documented return types are covered in a way that doesn't let anything else slip by unnoticed.

If we were to change the return type of `retrieve_password()`, then this change may be needed, but the return type is still `true|WP_Error` and the previous code here sufficiently covers these.